pkg/cmd/api/fields_test.go

```diff
@@ -61,8 +61,14 @@ func Test_parseFields_nested(t *testing.T) {
 			"robots[]=Dependabot",
 			"labels[][name]=bug",
 			"labels[][color]=red",
+			"labels[][colorOptions][]=red",
+			"labels[][colorOptions][]=blue",
 			"labels[][name]=feature",
 			"labels[][color]=green",
+			"labels[][colorOptions][]=red",
+			"labels[][colorOptions][]=green",
+			"labels[][colorOptions][]=yellow",
+			"nested[][key1][key2][key3]=value",
```

Comment:
How should the `gh api` command documentation be enhanced to demonstrate this new feature?

https://github.com/cli/cli/blob/234e0b3c8a332ab1b95595847ac615e315254d98/pkg/cmd/api/api.go#L67-L170

Reply:
Sorry for the delay on this!

How does this look? I dug through our REST examples and couldn't find many occurances of deeply nested arrays outside of custom properties (the endpoints that triggered this change), so I used them as the example: https://github.com/cli/cli/pull/8762/commits/3c3bed779f54135d01de735d9128d8157b2c43f8

Included the `default_value` because the endpoint will error if one is not specified when `allowed_values` are set

Reply:
@Ebonsignori : here are an assortment of REST endpoints that have `array of objects` parameters too:

- [Create an organization repository ruleset](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/rules?apiVersion=2022-11-28#create-an-organization-repository-ruleset)
- [Update an organization repository ruleset](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/rules?apiVersion=2022-11-28#update-an-organization-repository-ruleset)
- [Create or update IdP group connections](https://docs.github.com/en/enterprise-cloud@latest/rest/teams/team-sync?apiVersion=2022-11-28#create-or-update-idp-group-connections)
- [Create or update IdP group connections (Legacy)](https://docs.github.com/en/enterprise-cloud@latest/rest/teams/team-sync?apiVersion=2022-11-28#create-or-update-idp-group-connections-legacy)
- [Create a check run](https://docs.github.com/en/enterprise-cloud@latest/rest/checks/runs?apiVersion=2022-11-28#create-a-check-run)
- [Create a tree](https://docs.github.com/en/enterprise-cloud@latest/rest/git/trees?apiVersion=2022-11-28#create-a-tree)
- [Create or update an environment](https://docs.github.com/en/enterprise-cloud@latest/rest/deployments/environments?apiVersion=2022-11-28#create-or-update-an-environment)

Reply:
Beyond REST endpoints, there are many mutations that involve objects, so wondering what is an easy endpoint there to test these changes against

Reply:
Okay, here's me trying to test this out against GraphQL mutations that uses objects:

```shell
QUERY='
mutation($input:CreateRepositoryRulesetInput!){
  createRepositoryRuleset(input:$input){
    ruleset{
      id
      name
      createdAt
    }
  }
}
'

GH_DEBUG=api ./bin/gh api graphql -f query="$QUERY" \
	-F 'input[name]=test repo rule name3' \
	-F 'input[enforcement]=DISABLED' \
	-F 'input[conditions][refName][include][]=~DEFAULT_BRANCH' \
	-F 'input[conditions][refName][exclude][]=refs/heads/ignore-*' \
	-F 'input[sourceId]=R_kgDOIVGotg' \
	-F 'input[rules][][type]=CREATION'
```

with the resulting output:

```shell
$ ./test-nested-fields.sh
* Request at 2024-04-01 17:43:07.363237 -0400 EDT m=+0.049721293
* Request to https://api.github.com/graphql
> POST /graphql HTTP/1.1
> Host: api.github.com
> Accept: */*
> Authorization: token ████████████████████
> Content-Length: 399
> Content-Type: application/json; charset=utf-8
> Time-Zone: America/New_York
> User-Agent: GitHub CLI v2.44.1-6-g3c3bed77

GraphQL query:
mutation($input:CreateRepositoryRulesetInput!){
  createRepositoryRuleset(input:$input){
    ruleset{
      id
      name
      createdAt
    }
  }
}
GraphQL variables: {"input":{"conditions":{"refName":{"exclude":["refs/heads/ignore-*"],"include":["~DEFAULT_BRANCH"]}},"enforcement":"DISABLED","name":"test repo rule name3","rules":[{"type":"CREATION"}],"sourceId":"R_kgDOIVGotg"}}

< HTTP/2.0 200 OK
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
< Content-Security-Policy: default-src 'none'
< Content-Type: application/json; charset=utf-8
< Date: Mon, 01 Apr 2024 21:43:07 GMT
< Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
< Server: GitHub.com
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< Vary: Accept-Encoding, Accept, X-Requested-With
< X-Accepted-Oauth-Scopes: repo
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-Github-Media-Type: github.v4; format=json
< X-Github-Request-Id: E6F3:32CE1E:13928940:1F16DBCD:660B2A6B
< X-Oauth-Client-Id: 178c6fc778ccc68e1d6a
< X-Oauth-Scopes: admin:enterprise, gist, read:org, repo, workflow
< X-Ratelimit-Limit: 5000
< X-Ratelimit-Remaining: 5000
< X-Ratelimit-Reset: 1712011278
< X-Ratelimit-Resource: graphql
< X-Ratelimit-Used: 16
< X-Xss-Protection: 0

{
  "data": {
    "createRepositoryRuleset": {
      "ruleset": {
        "id": "RRS_lACqUmVwb3NpdG9yec4hUai2zgAJD8A",
        "name": "test repo rule name3",
        "createdAt": "2024-04-01T21:43:07Z"
      }
    }
  }
}

* Request took 285.7015ms
{
  "data": {
    "createRepositoryRuleset": {
      "ruleset": {
        "id": "RRS_lACqUmVwb3NpdG9yec4hUai2zgAJD8A",
        "name": "test repo rule name3",
        "createdAt": "2024-04-01T21:43:07Z"
      }
    }
  }
}
```